component/src/app/mod.rs

```diff
@@ -62,9 +64,34 @@ impl App {
         Dex::init_chain(&mut state_tx, app_state).await;
         StubDex::init_chain(&mut state_tx, app_state).await;
         Governance::init_chain(&mut state_tx, app_state).await;
-        // Shielded pool always executes last.
         ShieldedPool::init_chain(&mut state_tx, app_state).await;
 
+        let mut compact_block = state_tx.stub_compact_block();
+        let mut state_commitment_tree = state_tx.stub_state_commitment_tree().await;
+
+        // Hard-coded to zero because we are in the genesis block
+        // Tendermint starts blocks at 1, so this is a "phantom" compact block
+        compact_block.height = 0;
+
+        // Add current FMD parameters to the initial block.
+        compact_block.fmd_parameters = Some(state_tx.get_current_fmd_parameters().await.unwrap());
+
+        // Close the genesis block
+        state_tx
+            .finish_sct_block(&mut compact_block, &mut state_commitment_tree)
+            .await;
+
+        state_tx.set_compact_block(compact_block.clone());
+
+        state_tx
+            .write_sct(
+                compact_block.height,
+                state_commitment_tree,
+                compact_block.block_root,
+                compact_block.epoch_root,
+            )
+            .await;
```

Comment:
What's different about this logic compared to `finish_block` below?

Reply:
`finish_block` uses the current blockheight instead of hardcoding 0, and updates the chain parameters if they have changed.

Reply:
since in the future we'll want to support non-1 genesis heights, what if we set the height to `0` in `init_chain` and then reused the `finish_block` logic, letting it bundle the chain parameters into the initial compact block?

Reply:
Since we already `put_chain_params` and `put_block_height` in initchain, we can just replace this with finish_block